pkg/deploy/generator/resources_dev.go

```diff
@@ -397,7 +400,7 @@ func (g *generator) devDiskEncryptionSet() *arm.Resource {
 		Name:     to.StringPtr(fmt.Sprintf("[%s]", sharedDiskEncryptionSetName)),
 		Type:     to.StringPtr("Microsoft.Compute/diskEncryptionSets"),
 		Location: to.StringPtr("[resourceGroup().location]"),
-		Identity: &mgmtcompute.EncryptionSetIdentity{Type: mgmtcompute.SystemAssigned},
+		Identity: &mgmtcompute.EncryptionSetIdentity{Type: mgmtcompute.DiskEncryptionSetIdentityType(mgmtcompute.ResourceIdentityTypeSystemAssigned)},
```

Comment:
We may want to use `DiskEncryptionSetIdentityTypeSystemAssigned` instead of `ResourceIdentityTypeSystemAssigned`?

https://pkg.go.dev/github.com/Azure/azure-sdk-for-go@v68.0.0+incompatible/services/compute/mgmt/2021-12-01/compute#DiskEncryptionSetIdentityType

Reply:
`mgmtcompute.ResourceIdentityTypeSystemAssigned` evaluates to "SystemAssigned" so I think this is functionally equivalent to `mgmtcompute.DiskEncryptionSetIdentityType.DiskEncryptionSetIdentityTypeSystemAssigned` but I agree with you that just `mgmtcompute.DiskEncryptionSetIdentityType.DiskEncryptionSetIdentityTypeSystemAssigned` is better

Ref: 
https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources#ResourceIdentityType

Reply:
@nwnt If you can give your opinion on this I have a new joiner in the team who would love to pickup small tasks like this and get it over the line

Reply:
@mociarain sorry forgot to reply, I have made the changes as suggested.  can you give it another review please?